docs/ADR/2019-11-27-firestore-saving-timed-transcripts.md

```diff
@@ -0,0 +1,227 @@
+
+1. Easy to reason around
+2. Keeping cost down
+3. Firestore document size limit of 1mb 
+4. In Firestore reads of documents are charged individually 
+5. Ideally a solution that can scale for storing over 1 hour worth of transcription without breaking a sweat
+6. if needed paragraphs and words attribute could be saved in separate collections/documents.
+
+
+## Considered Options
+
+1. Break down data in firestore - "paginated" every 20 min
+2. Break down data in firestore - "paginated" every paragraph
+3. Break down data in firestore - every word is a document
+4. Saving json in Firebase cloud storage?
+5. Compression - JSONB 
+6. Compression - JSONC
+7. Compression - binary with UBJSON
+
+## Decision Outcome
+
+_TBC_
+<!-- Chosen option: "[option 1]", because [justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)]. -->
+
+<!-- ### Positive Consequences 
+
+* [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
+* …
+
+### Negative consequences
+
+* [e.g., compromising quality attribute, follow-up decisions required, …]
+* … -->
+
+## Pros and Cons of the Options <!-- optional -->
+
+### 1. Break down data in firestore - "paginated" every 20 min
+
+
+Data could be "paginated" in firestore, so that there could be a collection of words, and every x number of words (eg roughly every 3581 words objects), corresponding to 20 min worth of word objects could be split into a new document, eg words fragment.
+
+This means that one hour for instance would only need to be split into a collection of 3 documents.
+
+Two hours would be split into a collection of 6 documents, and so on.
+
+When saving it could be splitted into these documents, and when retrieving could be combined back into one data structure. 
+
+This is to show that the cost of the extra reads per transcription, would not be outrageous.
+
+A [fireabse HTTPS callable functions](https://firebase.google.com/docs/functions/callable) could be used to save and retrieve from firestore to do this fragmenting and aggregating.
+
+Using a cloud function as way to perform CRUD operations, would also incur the extra cost for the use of the cloud function. 
+
+Also need to double check if this would introduce a considerable delay in the user experience, as the function might need to boot up before performing the operation (?).
+
+* Good, because cost of reads would be contained
+* Bad, because introduces cost of calling a firebase cloud function to perform the operation
+* … <!-- numbers of pros and cons can vary -->
+
+
+
+### 2. Break down data in firestore - "paginated" every paragraph
+
+An alternative could also be to group the words by paragraphs.
+
+However when correcting paragraphs on the client, these might change and get re-arranged, so there might not be a lot of value in this. Eg when saving back to firestore might be easier to override everything then diff changes.
+
```

Comment:
Do you have an example of how many paragraphs would be in that 20 minute content?
I've looked at the sample document from what you send (2MB) file, I can see that there are 175 "alternatives", which might correspond to a paragraph.

Reply:
ah, no that's the wrong data structure, that is the GCP STT, which contains alternatives. DPE json is like this. See the Soleio interview in DPE demo linked at the top of the markdown.